operator/src/main/java/com/github/streamshub/console/dependents/ConsoleDeployment.java

```diff
@@ -82,9 +93,13 @@ protected Deployment desired(Console primary, Context<Console> context) {
                                 .withSecretName(configSecretName)
                             .endSecret()
                         .endVolume()
+                        .addAllToVolumes(getResourcesByType(trustResources, Volume.class))
                         .editMatchingContainer(c -> "console-api".equals(c.getName()))
                             .withImage(imageAPI)
-                            .addAllToEnv(coalesce(primary.getSpec().getEnv(), Collections::emptyList))
+                            .addAllToVolumeMounts(getResourcesByType(trustResources, VolumeMount.class))
+                            // Remove first to avoid duplicates
+                            .removeMatchingFromEnv(env -> envVars.stream().anyMatch(e -> env.getName().equals(e.getName())))
```

Comment:
the approach looks worrying.   doesn't this potentially overwrite env vars coming from the `console.deployment.yaml` with values provided by the user from the CR.

Reply:
That's a good point, the ENVs in the operator should always be present. If user attempts to override it results in a deployment error because they are duplicates. That is probably the correct behavior, but it happened to me trying to debug this and I added this remove line in haste. I'll go ahead and take it out.